mapper/tedge_mapper/src/mapper.rs

```diff
@@ -86,36 +91,108 @@ impl Mapper {
 
     #[instrument(skip(self), name = "messages")]
     async fn subscribe_messages(&self) -> Result<(), MqttClientError> {
-        let mut messages = self.client.subscribe(self.config.in_topic.filter()).await?;
+        let mut messages = self
+            .client
+            .subscribe(self.config.in_topic_filter.clone())
+            .await?;
+
+        let mut children: HashSet<String> = HashSet::new();
+
         while let Some(message) = messages.next().await {
             debug!("Mapping {:?}", message.payload_str());
-            match self.converter.convert(message.payload_str()?) {
-                Ok(mapped) => {
-                    self.client
-                        .publish(mqtt_client::Message::new(&self.config.out_topic, mapped))
-                        .await?;
-                }
-                Err(error) => {
-                    debug!("Mapping error: {}", error);
+            debug!("Topic {:?}", message.topic.name);
+
+            // ? Validation for child device ID?
+            match get_child_id_from_topic(message.clone().topic.name) {
+                Some(child_id) if child_id.is_empty() => {
```

Comment:
I might be easier to move the validation of the child id inside the `get_child_id_from_topic` function.

```
fn get_child_id_from_topic(topic: String) -> Result<Option<String>, ConversionError>
```